extensions/VisualEditor/wikia/modules/ve/dm/ve.dm.WikiaTransclusionModel.js

```diff
@@ -91,22 +92,47 @@ ve.dm.WikiaTransclusionModel.prototype.fetchInfoboxParamsRequestDone = function
 				};
 			}
 
-			for ( i = 0; i < page.infoboxes.length; i++ ) {
-				page.infoboxes[i].sources = Object.keys( page.infoboxes[i].sourcelabels );
-
-				for ( j = 0; j < page.infoboxes[i].sources.length; j++ ) {
-					source = page.infoboxes[i].sources[j];
-
-					specs[page.title].params[ source ] = { label: page.infoboxes[i].sourcelabels[source] };
-					specs[page.title].paramOrder.push( page.infoboxes[i].sources[j] );
-				}
-			}
+			page.infoboxes.forEach( function ( infobox ) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Define this function outside of a loop. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AFunctionDefinitionInsideLoop)